### PR TITLE
PSR-685 ETMP transformation of MP - member contributions - Get and Post

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

**Review by westwater — DISMISSED** [review]

File: app/controllers/nonsipp/membercontributions/MemberContributionsCYAController.scala [hunk 1]
```diff
@@ -57,26 +58,26 @@ class CYAMemberContributionsController @Inject()(
...
+                mode
               )
             )
           )
         )
       ).get
     }
 
-  def onSubmit(srn: Srn, checkOrChange: CheckOrChange): Action[AnyContent] =
+  def onSubmit(srn: Srn, mode: Mode): Action[AnyContent] =
```
Comment (westwater) [inline app/controllers/nonsipp/membercontributions/MemberContributionsCYAController.scala:75]: 👍

File: app/controllers/nonsipp/membercontributions/MemberContributionListController.scala [hunk 2]
```diff
@@ -26,74 +27,138 @@ import eu.timepit.refined.api.Refined
...
+      Future.fromTry(
+        indexes.fold(
+          _ => userAnswerWithMemberContList,
+          index =>
+            index.foldLeft(userAnswerWithMemberContList) {
+              case (uaTry, index) =>
+                val optTotalMemberContribution = request.userAnswers.get(TotalMemberContributionPage(srn, index))
+                for {
+                  ua <- uaTry
+                  ua1 <- ua.set(TotalMemberContributionPage(srn, index), optTotalMemberContribution.getOrElse(Money(0)))
```
Comment (kberec) [inline app/controllers/nonsipp/membercontributions/MemberContributionListController.scala:150]: I don't think we should set anything to 0 in the FE.
Let's have a chat about this @cyeniceri

Comment (Vamshi-Muniganti) [reply app/controllers/nonsipp/membercontributions/MemberContributionListController.scala:150]: its setting to 0 by default even we dont report any member contributions

**Review by kberec — APPROVED** [review]

File: app/controllers/nonsipp/membercontributions/MemberContributionListController.scala (hunk 2 above)
Comment (kberec) [reply app/controllers/nonsipp/membercontributions/MemberContributionListController.scala:150]: We had a chat and we'll discuss alternative solutions on the sprint review.

**Pull request merged**